pkg/codegen/schema/loader.go

```diff
@@ -210,3 +250,106 @@ func LoadPackageReference(loader Loader, pkg string, version *semver.Version) (P
 	}
 	return p.Reference(), nil
 }
+
+func (l *pluginLoader) loadSchemaBytes(pkg string, version *semver.Version) ([]byte, *semver.Version, error) {
+	pluginInfo, err := workspace.GetPluginInfo(workspace.ResourcePlugin, pkg, nil, false /* skipMetadata */)
```

Comment:
I'm a bit surprised by this. Is this to avoid the overhead of loading the provider plugin? If we loaded the plugin, we could just ask it for its info. I'm interested primarily because I don't love letting the `workspace` concept leak into the package loader--I'd much rather leave that logic for the plugin host. If we loaded the provider plugin, I think we could ask it for its info directly via `GetPluginInfo`.

Reply:
This was so that we could get the plugin version & directory, otherwise when the version is nil (quite common for YAML and our tests) and we lack a directory, we don't have a unique name for the cache file. It leads to some peculiar errors when using the EKS provider (uses aws v4) and the awsx provider (uses aws v5).

I've refactored this to a `ResolvePlugin` method on the `plugin/host.go Host` interface to support `deploytest`. 

This ensures we put the plugin in a versioned directory alongside the plugin executable without having to host it.

Reply:
I've pushed several of the implementation details of the filesystem into fields on PluginInfo returned from this method. Schema Loader is now agnostic to the filesystem details and lets the host determine where and how to cache.